main/src/com/google/refine/operations/row/RowDuplicatesRemovalOperation.java

```diff
@@ -0,0 +1,108 @@
+/*
+
+Copyright 2024 OpenRefine
+All rights reserved.
+*/
+
+package com.google.refine.operations.row;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.AbstractOperation;
+import com.google.refine.model.Column;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.model.changes.RowRemovalChange;
+import com.google.refine.operations.OperationDescription;
+
+public class RowDuplicatesRemovalOperation extends AbstractOperation {
+
+    final protected List<String> _criteria;
+
+    @JsonCreator
+    public RowDuplicatesRemovalOperation(
+            @JsonProperty("criteria") List<String> criteria) {
+        _criteria = criteria;
+    }
+
+    @JsonProperty("criteria")
+    public List<String> getRows() {
+        return _criteria;
+    }
+
+    @Override
+    protected String getBriefDescription(Project project) {
+        return OperationDescription.row_remove_duplicates_brief();
+    }
+
+    @Override
+    protected HistoryEntry createHistoryEntry(Project project, long historyEntryID) throws Exception {
+
+        List<Integer> rowIndices = new ArrayList<Integer>();
+        final List<Column> criteriaColumns = new ArrayList<Column>(_criteria.size());
+        for (String c : _criteria) {
+            Column toColumn = project.columnModel.getColumnByName(c);
+            if (toColumn != null) {
+                criteriaColumns.add(toColumn);
+            }
+        }
+
+        findDuplicateRows(project, criteriaColumns, rowIndices);
+
+        return new HistoryEntry(
+                historyEntryID,
+                project,
+                "Duplicate rows removal completed. " + rowIndices.size() + " rows removed.",
```

Comment:
The `completed.` part strikes me as inconsistent with the style of the other history entry descriptions. Perhaps `Remove 123 duplicate rows.` would be simpler and more fitting (see for instance `RowRemovalOperation`).

Reply:
Updated